### PR TITLE
[Merged by Bors] - feat: port Data.Nat.ModEq

### PR DESCRIPTION
**Pull request by hargoniX**

### CODE REVIEW

File: Mathlib/Data/Nat/ModEq.lean [hunk 1]
```diff
@@ -0,0 +1,532 @@
...
+instance : Trans (ModEq n) (ModEq n) (ModEq n) where
+  trans := Nat.ModEq.trans
+
+protected theorem comm : a ≡ b [MOD n] ↔ b ≡ a [MOD n] :=
+  ⟨ModEq.symm, ModEq.symm⟩
+#align nat.ModEq.comm Nat.ModEq.comm
+
+end ModEq
+
+theorem ModEq_zero_iff_dvd : a ≡ 0 [MOD n] ↔ n ∣ a := by rw [ModEq, zero_mod, dvd_iff_mod_eq_zero]
```
Comment (hrmacbeth) [inline Mathlib/Data/Nat/ModEq.lean:82]: Just a drive-by comment because I'm not an expert mathlib4 reviewer yet.  Here and in similar cases, I think the name should be `modEq_zero_iff_dvd`, according to [naming convention](https://github.com/leanprover-community/mathlib4/wiki/Porting-wiki#naming-convention) rule 5:
> When something named with `UpperCamelCase` is part of something named with `snake_case`, it is referenced in `lowerCamelCase`.

One could even argue that it should be `modeq_zero_iff_dvd` here, according to rule 6:
> Acronyms like LE are written upper-/lowercase as a group, depending on what the first character would be.

File: Mathlib/Data/Nat/ModEq.lean [hunk 2]
```diff
@@ -0,0 +1,532 @@
...
+notation:50 a " ≡ " b " [MOD " n "]" => ModEq n a b
+
+variable {m n a b c d : ℕ}
+
+-- Porting note: This instance should be derivable automatically
+instance : Decidable (ModEq n a b) :=
+  if h : a % n = b % n then
+    isTrue h
+  else
+    isFalse h
```
Comment (hrmacbeth) [inline Mathlib/Data/Nat/ModEq.lean:48]:
```suggestion
instance : Decidable (ModEq n a b) := decEq (a % n) (b % n)
```

File: Mathlib/Data/Nat/ModEq.lean (hunk 1 above)
Comment (dupuisf) [reply Mathlib/Data/Nat/ModEq.lean:82]: I think `modEq` is the way to go here since we picked `ModEq` over `Modeq` for the prop.

Comment (hargoniX) [reply Mathlib/Data/Nat/ModEq.lean:78]: I think I got all fixed, at least both my eyes and grep can't seem to find an exception but probably someone should take a look again just in case.

**Comment by hargoniX** [comment]
I'm not quite sure I understand why it is failing? I rebased on master and it is building locally for me.

File: Mathlib/Data/Nat/ModEq.lean [hunk 3]
```diff
@@ -0,0 +1,528 @@
...
+-/
+import Mathlib.Data.Int.GCD
+import Mathlib.Data.Int.Order.Lemmas
+
+/-!
+# Congruences modulo a natural number
+
+This file defines the equivalence relation `a ≡ b [MOD n]` on the natural numbers,
+and proves basic properties about it such as the Chinese Remainder Theorem
+`ModEq_and_ModEq_iff_ModEq_mul`.
```
Comment (dupuisf) [inline Mathlib/Data/Nat/ModEq.lean:19]:
```suggestion
`modEq_and_modEq_iff_modEq_mul`.
```

**Review by dupuisf — COMMENTED** [review]
Just one left in the docstring -- thanks!

bors d+
